### ISSUE TITLE
really needs Message-ID header

### ISSUE THREAD

**Issue by jhaar** [gophish/gophish#1417]
What version of Gophish are you using?: 

current binary for Linux (0.7.1).

Brief description of the issue: 

Being caught as spam: gophish doesn't add a Message-ID header - which is a pseudo-requirement according to RFC5322. Most antispam solutions will mark down email without it - and Google even mentions it WRT delivering email to Gmail

https://support.google.com/a/answer/81126

What are you expecting to see happen? :

Have a Message-ID header

What are you seeing happen?

Have a Message-ID header

I actually tried faking one using the "Additional headers" feature, but I of course each email will needs it's own unique Message-ID header, and my attempts at "<{{.Rid}}{{.FirstName}}{{.LastName}}@example.com>" created a Message-ID of "<" (ie one char) - so that went wrong

It would be better to do a good generalized header that looks something like "Message-ID: <523B6001.3010003@example.com>"

**Comment by jordan-wright** [comment]
Hi @jhaar,

This should be no problem to add. I guess the gomail library doesn't add a message ID (I thought it did).

Thanks for the heads up!

**Comment by jhaar** [comment]
Actually, I found a workaround. I added a header, ie

Message-ID with value "<randomstring-{{.RId}}@our.domain>" 

(don't forget the angle brackets).

That is enough to emulate a legitimate and unique-per-msg Message-ID, with the additional advantage I can make "our.domain" match or not match what I'm using in the From: field (which can add favour to testing). 

Hint for beginners like me: it's "{{.RId}}" - not ".{{Rid}}" (uppercase I) - I had tried this unsuccessfully earlier. :-/

**Comment by justaskingquestions** [comment]
@jhaar  Thanks for the hint with the message id! My e-mails to G-Mail started bouncing compared to last year. This fixed it.

**Comment by aczietlow** [comment]
I took a crack at adding the message-id header by default.

**Comment by jordan-wright** [comment]
Closed via #1441

**Comment by justaskingquestions** [comment]
I know this is fixed now, but AFAIK not in any release build yet. 
I have found an additional solution for those who are not running the dev build and are using postifx as a mail server Adding the following line to your main.cf and restarting postfix will add a vaild Message-ID header matching the hostname of your mailserver.
```
#Interim fix to add message-id header. This should be obsolete with gophish 0.8
always_add_missing_headers = yes
```
